tracer/test/test-applications/integrations/Samples.RabbitMQ/Program.cs

```diff
@@ -220,23 +223,26 @@ private static void Receive(bool useQueue)
 
                 if (useQueue)
                 {
-                    DequeueThread.Start();
+                    _dequeueThread = new Thread(() => ConsumeFromQueue(queue));
+                    _dequeueThread.Start();
                 }
 
-                while (_messageCount != 0)
+                while (Interlocked.Read(ref _messageCount) != 0)
                 {
                     Thread.Sleep(100);
                 }
 
+                queue.CompleteAdding();
```

Comment:
I think it'd be cleaner if you waited for the DequeueThread here instead of letting RunRabbitMQ doing it. Basically, hide entirely the fact that Receive is using a thread.

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/2736/commits/264059a6144006ce866b77bade9246bdf8910806